lib/slide_hub/cloud/engine/azure.rb

```diff
@@ -0,0 +1,163 @@
+
+        def self.config
+          @config ||= Config.new
+        end
+
+        def self.queue_endpoint
+          url = "https://#{@config.azure_storage_account_name}.queue.core.windows.net/"
+          url
+        end
+
+        def self.resource_endpoint
+          return "#{@config.cdn_base_url}/#{@config.image_bucket_name}" unless @config.cdn_base_url.blank?
+
+          "https://#{@config.azure_storage_account_name}.blob.core.windows.net/#{@config.image_bucket_name}"
+        end
+
+        def self.upload_endpoint
+          url = "https://#{@config.azure_storage_account_name}.blob.core.windows.net/#{@config.bucket_name}"
+          url
+        end
+
+        def self.send_message(message)
+          queues = ::Azure.queues
+          resp = queues.create_message(@config.queue_name, message)
+          resp
+        end
+
+        def self.receive_message(max_number = 10)
+          queues = ::Azure.queues
+          response = queues.list_messages(@config.queue_name, 600, { number_of_messages: max_number })
+          # see http://www.rubydoc.info/github/Azure/azure-sdk-for-ruby/Azure%2FQueue%2FQueueService%3Alist_messages
+          # it returns a list of Azure::Entity::Queue::Message instances
+          # http://www.rubydoc.info/github/Azure/azure-sdk-for-ruby/Azure/Queue/Message
+          result = SlideHub::Cloud::Queue::Response.new
+          unless (!response || response.count == 0)
+            response.each do |msg|
+              result.add_message(msg.id, msg.message_text, msg.pop_receipt)
+            end
+          end
+          result
+        end
+
+        def self.delete_message(message_object)
+          queues = ::Azure.queues
+          queues.create_queue(@config.queue_name)
+          resp = queues.delete_message(@config.queue_name, message_object.id, message_object.handle)
+          resp
+        end
+
+        def self.batch_delete(entries)
+          result = true
+          entries.each do |entry|
+            result = self.delete_message(entry)
+          end
+          result
+        end
+
+        ## Blob
+        def self.upload_files(container, files, prefix)
+          bs = ::Azure::Blob::BlobService.new
+          files.each do |f|
+            if File.exist?(f)
+              content = File.open(f, 'rb', &:read)
+              bs.create_block_blob(container, "#{prefix}/#{File.basename(f)}", content)
+            end
+          end
+        end
+
+        def self.get_file_list(container, prefix)
+          bs = ::Azure::Blob::BlobService.new
+          resp = bs.list_blobs(container, { prefix: prefix, max_results: 1000 })
+          files = []
+          resp.each do |blob|
+            files.push({ key: blob.name })
+          end
+          files
+        end
+
+        def self.save_file(container, key, destination)
+          bs = ::Azure::Blob::BlobService.new
+          _blob, content = bs.get_blob(container, key)
+          File.open(destination, 'wb') { |f| f.write(content) }
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'f' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)